compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -401,6 +401,8 @@ object Checking {
       if (!sym.is(Deferred))
         fail(NativeMembersMayNotHaveImplementation(sym))
     }
+    if (sym.hasAnnotation(defn.CompileTimeOnlyParamAnnot))
+      ctx.migrationWarning("`@compileTimeOnly(msg)` will be replaced by `scala.compiletime.error(msg)` or `erased`", sym.sourcePos)
```

Comment:
Maybe remove `or erased`, which is not an obvious replacement -- where to put the message?

Reply:
Good point